src/librustc_typeck/check/expr.rs

```diff
@@ -566,43 +566,48 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             // the `enclosing_loops` field and let's coerce the
             // type of `expr_opt` into what is expected.
             let mut enclosing_breakables = self.enclosing_breakables.borrow_mut();
-            let ctxt = enclosing_breakables.find_breakable(target_id);
-            if let Some(ref mut coerce) = ctxt.coerce {
-                if let Some(ref e) = expr_opt {
-                    coerce.coerce(self, &cause, e, e_ty);
+            if let Some(ref mut ctxt) = enclosing_breakables.opt_find_breakable(target_id) {
```

Comment:
I think this could use a comment wrt. why it was `None` in the first place.
Also -- should this delay_span_bug in the `else` case that isn't included?